opentracing-spring-cloud-starter-jaeger/src/main/java/io/opentracing/contrib/spring/cloud/starter/jaeger/JaegerAutoConfiguration.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    private Reporter getHttpReporter(Metrics metrics,
+        JaegerConfigurationProperties.RemoteReporterProperties remoteReporterProperties,
+        JaegerConfigurationProperties.HttpSender httpSenderProperties) {
+      com.uber.jaeger.senders.HttpSender httpSender = new com.uber.jaeger.senders.HttpSender(
+          httpSenderProperties.getUrl(), httpSenderProperties.getMaxPayload());
+      return new RemoteReporter(httpSender, remoteReporterProperties.getFlushInterval(),
+          remoteReporterProperties.getMaxQueueSize(), metrics);
+    }
+
+    @ConditionalOnMissingBean
+    @Bean
+    public Metrics reporterMetrics(StatsReporter statsReporter) {
+      return Metrics.fromStatsReporter(statsReporter);
+    }
+
+    @ConditionalOnMissingBean
+    @Bean
+    public StatsReporter statsReporter(JaegerConfigurationProperties properties) {
+      if (properties.isEnableMetrics()) {
+        return new InMemoryStatsReporter();
+      }
+      return new NullStatsReporter();
+    }
+
+    @ConditionalOnProperty(value = "opentracing.jaeger.enableB3Propagation", havingValue = "true")
+    @Bean
+    public JaegerTracerCustomizer b3CodecJaegerTracerCustomizer() {
+      return new B3CodecJaegerTracerCustomizer();
+    }
+
+    /**
+     * Decide on what Sampler to use based on the various configuration options in
+     * JaegerConfigurationProperties Fallback to ConstSampler(true) when no Sampler is configured
+     */
+    @ConditionalOnMissingBean
+    @Bean
+    public Sampler sampler(JaegerConfigurationProperties properties, Metrics metrics) {
+      if (properties.getConstSampler().getDecision() != null) {
+        return new ConstSampler(properties.getConstSampler().getDecision());
+      }
+
+      if (properties.getProbabilisticSampler().getSamplingRate() != null) {
+        return new ProbabilisticSampler(properties.getProbabilisticSampler().getSamplingRate());
+      }
+
+      if (properties.getRateLimitingSampler().getMaxTracesPerSecond() != null) {
+        return new RateLimitingSampler(properties.getRateLimitingSampler().getMaxTracesPerSecond());
+      }
+
+      if (!StringUtils.isEmpty(properties.getRemoteControlledSampler().getHostPort())) {
+        JaegerConfigurationProperties.RemoteControlledSampler samplerProperties
+            = properties.getRemoteControlledSampler();
+
+        Sampler initialSampler = new ProbabilisticSampler(samplerProperties.getSamplingRate());
+        HttpSamplingManager manager = new HttpSamplingManager(samplerProperties.getHostPort());
+
+        return new RemoteControlledSampler(properties.getServiceName(), manager, initialSampler,
+            metrics);
+      }
+
+      //fallback to sampling every trace
+      return new ConstSampler(true);
+    }
+
+  }
+
+
+  @Configuration
+  @ConditionalOnProperty(value = "opentracing.jaeger.useTracerResolver", havingValue = "true")
+  @ConditionalOnClass(JaegerTracerResolver.class)
+  public static class TracerResolverConfiguration {
```

Comment:
See the previous comments for tracerresolver

Reply:
https://github.com/redhat-developer-demos/istio-tutorial/blob/master/customer/java/pom.xml#L46

Reply:
We could perhaps remove support for the TracerResolver configuration altogether... I only added it to take advantage of the way Spring Boot resolves properties, which is more flexible than just system properties or environment variables.
I don't expect people will actually use this feature of the auto-configuration module, so we could just remove it...

Reply:
What is the default behaviour if the jaeger starter is on classapath? Does it create jaeger tracer with default configuration or does ot fall back to tracer resolver?

Reply:
The default behavior is to create the Jaeger tracer with some "sensible" default settings

Reply:
I am still not sure if we want to have `TracerResolverConfiguration` as the same behavior can be achieved by using ot-spring-cloud with jaeger-tracerresolver on classpath. I am more -1 on adding this.

@objectiser @jpkrohling any opinion?

Reply:
@pavolloffay I would be more than happy to remove it :)

Reply:
Go ahead then :)

Reply:
Done :)